.github/workflows/deploy_staging.yml

```diff
@@ -36,7 +37,10 @@ jobs:
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ${{ env.AWS_REGION }}
+          aws-region: ${{ vars.STAGING_AWS_REGION }}
 
       - name: Sync to S3
-        run: aws s3 sync ./dist s3://${{ env.S3_BUCKET }} --delete
+        run: aws s3 sync ./dist s3://${{ vars.STAGING_S3_BUCKET }} --delete
+
+      - name: Invalidate CloudFront cache
+        run: aws cloudfront create-invalidation --distribution-id ${{ vars.STAGING_CLOUDFRONT_DISTRIBUTION_ID }} --paths "/*"
```

Comment:
```
# function to wait for invalidation to complete with distribution id as parameters
function wait_for_invalidation {
    STATUS="InProgress"
    INVALIDATION_ID=$(aws cloudfront create-invalidation  --distribution-id "$1" --paths "/*" | jq --raw-output '.Invalidation.Id')
    echo "Waiting for invalidation $INVALIDATION_ID to complete... on $1"
    while [ "$STATUS" != "Completed" ]; do
        STATUS=$(aws cloudfront get-invalidation --id $INVALIDATION_ID --distribution-id $1 | jq --raw-output '.Invalidation.Status')
        sleep 5
        echo "Invalidation Status: $STATUS"
    done
}
```

it can be handy to keep the job running until the invalidation completes

Reply:
of course then you need jq on the build runner :/

Reply:
Done, example: https://github.com/MoneyBadgers/moneybadger-payments-page/actions/runs/9414739581/job/25934251821